source/Nuke.Common/Tooling/PackageExecutableAttribute.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Maintainers of NUKE.
+// Distributed under the MIT License.
+// https://github.com/nuke-build/nuke/blob/master/LICENSE
+
+using System;
+using System.Linq;
+using Nuke.Common.Execution;
+
+namespace Nuke.Common.Tooling
+{
+    /// <inheritdoc/>
+    /// <summary>
+    ///     Injects a delegate for process execution. The path to the executable is resolved in the following order:
+    ///     <ul>
+    ///         <li>From environment variables (e.g., <c>[MEMBER_NAME]_EXE=path</c>)</li>
+    ///         <li>
+    ///             From the <c>.csproj</c> or <c>packages.config</c> file using <c>packageId</c> and
+    ///             <c>packageExecutable</c> to look up the NuGet package cache
+    ///         </li>
+    ///     </ul>
+    ///     <inheritdoc/>
+    /// </summary>
+    /// <example>
+    ///     <code>
+    /// [PackageExecutable("NuGet.CommandLine", "nuget.exe")] readonly Tool NuGet;
+    /// Target FooBar => _ => _
+    ///     .Executes(() =>
+    ///     {
+    ///         var process = NuGet($"pack {ProjectFile}");
+    ///         process.AssertZeroExitCode();
+    ///     });
+    ///     </code>
+    /// </example>
+    public class PackageExecutableAttribute : InjectionAttributeBase
+    {
+        private readonly string _packageId;
+        private readonly string _packageExecutable;
+        private readonly string _framework;
+
+        public PackageExecutableAttribute(string packageId, string packageExecutable, string framework = null)
+        {
+            _packageId = packageId;
+            _packageExecutable = packageExecutable;
+            _framework = framework;
+        }
+
+        public override object GetValue(string memberName, Type memberType)
+        {
+            var toolPath =
+                ToolPathResolver.TryGetEnvironmentExecutable($"{memberName.ToUpperInvariant()}_EXE") ??
```

Comment:
I think this is worth to discuss. For `PathExecutable` it kind of makes sense, but what about executables from NuGet packages? For instance, for `Xunit` the executable is called `xunit.console.exe`. Should we still allow to override via `XUNIT_EXE` ?